src/components/UsageBar.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+  <div class="header">
+    <slot name="header" />
+  </div>
+  <ProgressBar
+    v-for="(item, index) in items"
+    :badge-label="item.badgeLabel"
+    :key="index"
+    :label="item.label"
+    :max-value="item.maxValue"
+    :value="item.value"
+  />
+  <div class="footer">
+    <slot
+      name="footer"
+      :average-percent="averagePercent"
+      :total-value="totalValue"
+    />
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { computed } from "vue";
+import ProgressBar from "@/components/ProgressBar.vue";
+
+interface Data {
+  badgeLabel?: string;
+  label?: string;
+  maxValue?: number;
+  value: number;
+}
+
+interface Props {
+  data: Data[];
+  title?: string;
+}
+
+const props = defineProps<Props>();
+
+const items = computed(() =>
+  props.data
+    .map((item) => ({
+      ...item,
+      percent: Math.round((item.value / (item.maxValue ?? 100)) * 100),
+    }))
+    .sort((item, nextItem) => nextItem.percent - item.percent)
+);
+
+const averagePercent = computed(
+  () =>
+    items.value.reduce((total, item) => total + item.percent, 0) /
+    items.value.length
+);
+
+const totalValue = computed(() =>
+  items.value.reduce((total, item) => total + item.value, 0)
+);
+</script>
+
+<style scoped lang="postcss">
+.header {
+  border-bottom: 1px solid var(--color-extra-blue-base);
+  color: var(--color-extra-blue-base);
+  display: flex;
+  font-size: 16px;
+  font-weight: 700;
+  justify-content: space-between;
+  margin-bottom: 2rem;
+}
+.footer {
+  color: var(--color-blue-scale-300);
+  display: flex;
+  font-weight: 700;
+  font-size: 14px;
+  justify-content: space-between;
+}
+
+.progress-bar {
+  &:nth-of-type(2) :deep(.bar-fill),
+  &:nth-of-type(2) :deep(.circle) {
+    background-color: var(--color-extra-blue-d60);
+  }
+  &:nth-of-type(3) :deep(.bar-fill),
+  &:nth-of-type(3) :deep(.circle) {
+    background-color: var(--color-extra-blue-d40);
+  }
+  &:nth-of-type(4) :deep(.bar-fill),
+  &:nth-of-type(4) :deep(.circle) {
+    background-color: var(--color-extra-blue-d20);
+  }
```

Comment:
I don't get why it goes from 2 to 4 instead of 1 to 4.

Reply:
On Clémence's model, there are only 4 different colors in the gradient. All items after 3 have the same color.
We start at 2 because the `div` `header` is considered as 1

Reply:
You mean that `.progress-bar:nth-of-type(1)` matches the header? If so, I don't get why because the header doesn't have the `progress-bar` class.

Reply:
`.progress-bar:nth-of-type(1)` match nothing because there is no element as first child with `progress-bar` class.

Reply:
Ping @ByScripts, I can't close conversations, probably because it's your repo.